### PR TITLE
Fix MC-167561 - All players can make a dog stand up or sit

### PR DESCRIPTION
**Pull request by perseus5**

### CODE REVIEW

File: Spigot-Server-Patches/0440-Fix-MC-167561-All-players-can-make-a-dog-stand-up-or.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
+index db15d5e0a..83a515df4 100644
+--- a/src/main/java/net/minecraft/server/EntityWolf.java
++++ b/src/main/java/net/minecraft/server/EntityWolf.java
+@@ -295,7 +295,7 @@ public class EntityWolf extends EntityTameableAnimal {
+                 if (!(item instanceof ItemDye)) {
+                     boolean flag = super.a(entityhuman, enumhand);
+ 
+-                    if (!flag || this.isBaby()) {
++                    if ((!flag || this.isBaby()) && this.i(entityhuman)) {
```
Comment (mikroskeem) [inline Spigot-Server-Patches/0440-Fix-MC-167561-All-players-can-make-a-dog-stand-up-or.patch]: `// Paper - Fix MC-167561`

**Review by mikroskeem — CHANGES_REQUESTED** [review]

**Comment by BillyGalbreath** [comment]
To be completely honest, the best move here is to revert the entire method back to what it was on 1.14.4, I did that on purpur and _all_ wolf interaction bugs went away

File: Spigot-Server-Patches/0440-Fix-MC-167561-All-players-can-make-a-dog-stand-up-or.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
+index db15d5e0a..099ebb006 100644
+--- a/src/main/java/net/minecraft/server/EntityWolf.java
++++ b/src/main/java/net/minecraft/server/EntityWolf.java
+@@ -295,7 +295,7 @@ public class EntityWolf extends EntityTameableAnimal {
+                 if (!(item instanceof ItemDye)) {
+                     boolean flag = super.a(entityhuman, enumhand);
+ 
+-                    if (!flag || this.isBaby()) {
++                    if ((!flag || this.isBaby()) && this.i(entityhuman)) { // Paper - Fix MC-167561
```
Comment (BillyGalbreath) [inline Spigot-Server-Patches/0440-Fix-MC-167561-All-players-can-make-a-dog-stand-up-or.patch:17]: But, if reverting the entire method isn't in the core team's interest, at least make an obfhelper for the `isOwner` call here

**Review by BillyGalbreath — CHANGES_REQUESTED** [review]

**Comment by electronicboy** [comment]
Need to reference what mojang have opted to do in 1.16 to make a decision on this, I'm not 100% convinced that this is the proper fix, the behavior here needs comparing to what it "should" be

**Comment by kennytv** [comment]
They moved a check and execution up to that part, also including the isOwned check, it should look (exactly) like this (`this.i(itemstack)` checks if the item is food)
```
                if (!(item instanceof ItemDye)) {
                    boolean flag = super.a(entityhuman, enumhand);

                    if ((!flag || this.isBaby()) && this.i(entityhuman) && !this.i(itemstack)) {
                        this.goalSit.setSitting(!this.isSitting());
                        this.jumping = false;
                        this.navigation.o();
                        this.setGoalTarget((EntityLiving) null, TargetReason.FORGOT_TARGET, true); // CraftBukkit - reason
                    }

                    return flag;
                }
```
as well as removing that part from below

**Pull request closed without merging**